js/components/common/__tests__/CourseModalButtons.test.js

```diff
@@ -32,142 +25,118 @@ describe("CourseModal", () => {
   })
 
   it("enables back button when a course link is clicked", async () => {
-    await waitForModalUpdate()
-
     // click on the course link BBB100H1, and BBB100H1's modal opens
     const bbbCourseLink = screen.getByText("BBB100H1")
     fireEvent.click(bbbCourseLink)
 
-    await waitForModalUpdate()
-
     // the forward button is disabled, but the back button is enabled
-    const forwardButton = screen.getByText(">")
+    const forwardButton = await screen.findByText(">")
```

Comment:
So in all of these tests, I think it's better to call `act` on the `fireEvent` actions above, and leave these queries to be the synchronous version (i.e., `getByText`). This seems more consistent with others uses of `act` that you've added (consistency will also make it easier to replace these with something like [`user-event`](https://testing-library.com/docs/user-event/intro) in the future)